src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/RoleController.kt

```diff
@@ -37,6 +39,54 @@ class RoleController(context: ActionContext,
         return roleRepo.getAllRoleNames()
     }
 
+    fun addRole(): String {
+        val name = context.postData()["name"] ?: throw MissingParameterError("name")
+        authRepo.createUserGroup(name)
+        return okayResponse()
+    }
+
+    fun associatePermission(): String
```

Comment:
Bit of duplicate code here between the two controllers when wrangling post data - could go in base class or util?

Reply:
this whole method is split in 2 in the next PR, with shared logic